docs: add respect command docs (#1923)
Redocly/redocly-cli: docs/commands/respect.md

---
slug:
  - /docs/cli/commands/respect
  - /docs/respect/commands/respect
rbac:
  authenticated: read
---

# `respect`

> Use this command to execute API tests described in an Arazzo file.

Reviewer: ```suggestion
Use this command to execute API tests described in an Arazzo description.
```

In addition to the Arazzo specification, `respect` supports specification extensions for API testing: `x-operation` and `x-serverUrl`.

## Usage

```sh
npx @redocly/cli respect <your-test-file | multiple files | files bash query> [-w | --workflow] [-s | --skip] [-v | --verbose] [-i | --input]
```

## Options

{% table %}

- Option {% width="20%" %}
- Type {% width="15%" %}
- Description

---

- -w, --workflow
- [string]
- Workflow names from the test file to run.

>   For example, the following command runs "first-flow" and "second-flow" workflows from the "test-file.yaml" test file: `npx @redocly/cli respect test-file.yaml --workflow first-flow second-flow`

Reviewer: ```suggestion
  For example, the following command runs "first-flow" and "second-flow" workflows from the "test-file.yaml" Arazzo description: `npx @redocly/cli respect test-file.yaml --workflow first-flow second-flow`
```

  {% admonition type="warning" %}
  The `--workflow` option can't be used with `--skip`.
  {% /admonition %}

---

- -s, --skip
- [string]
- Workflow names from the test file to skip.

>   For example, the following command skips the "first-flow" workflow from the "test-file.yaml" test file: `npx @redocly/cli respect test-file.yaml --skip first-flow`

Reviewer: ```suggestion
  For example, the following command skips the "first-flow" workflow from the "test-file.yaml" Arazzo description: `npx @redocly/cli respect test-file.yaml --skip first-flow`
```

>   {% admonition type="warning" %}

Reviewer: ```suggestion
  {% admonition type="warning" name="Warning" %}
```

>   Warning: the `--skip` option can't be used with `--workflow`.

Reviewer: ```suggestion
  The `--skip` option can't be used with `--workflow`.
```

  {% /admonition %}

---

- -v, --verbose
- boolean
- Runs the command in verbose mode to help with troubleshooting issues.

>   For example, the following command runs all workflows from the "test-file.yaml" test file in verbose mode: `npx @redocly/cli respect test-file.yaml --verbose`

Reviewer: ```suggestion
  For example, the following command runs all workflows from the "test-file.yaml" Arazzo description in verbose mode: `npx @redocly/cli respect test-file.yaml --verbose`
```

---

- --har-output
- string
- Path for the `har` file for saving logs.

>   For example, the following command runs all workflows from the "test-file.yaml" test file and saves the logs to the "logs.har" file: `npx @redocly/cli respect test-file.yaml --har-output='logs.har'`

Reviewer: ```suggestion
  For example, the following command runs all workflows from the "test-file.yaml" Arazzo description and saves the logs to the "logs.har" file: `npx @redocly/cli respect test-file.yaml --har-output='logs.har'`
```

---

- --json-output
- string

> - Path for the 'json`file for saving logs. For example, the following command runs all workflows from the "test-file.yaml" test file and saves the logs to the "logs.json" file:`npx @redocly/cli respect test-file.yaml --json-output='logs.json'`

Reviewer: ```suggestion
- Path for the JSON file for saving logs. For example, the following command runs all workflows from the "test-file.yaml" Arazzo description and saves the logs to the "logs.json" file:`npx @redocly/cli respect test-file.yaml --json-output='logs.json'`
```

---

- --input
- string
- Input parameters with values that are mapped to the workflow inputs description.

>   For example, the following command maps the "userEmail" and "userPassword" inputs and values to all workflows in the "test.yaml" test file: `npx @redocly/cli respect test.yaml --input userEmail=name@redocly.com --input userPassword=12345`.

Reviewer: ```suggestion
  For example, the following command maps the "userEmail" and "userPassword" inputs and values to all workflows in the "test.yaml" Arazzo description: `npx @redocly/cli respect test.yaml --input userEmail=name@redocly.com --input userPassword=12345`.
```

  You can also use an environment variable to set the input, as in the following example: `REDOCLY_CLI_RESPECT_INPUT='userEmail=name@redocly.com,userPassword=12345' npm run cli respect test.yaml`

>   You can even include nested values, as in the following example command that maps the "nestedKey" input and value to all workflows in the "test-file.yaml" test file: `npx @redocly/cli respect test-file.yaml --input '{"key": "value", "nested": {"nestedKey": "nestedValue"}}'`.

Reviewer: ```suggestion
  You can even include nested values, as in the following example command that maps the "nestedKey" input and value to all workflows in the "test-file.yaml" Arazzo description: `npx @redocly/cli respect test-file.yaml --input '{"key": "value", "nested": {"nestedKey": "nestedValue"}}'`.
```

  You can also use an environment variable to set the input, as in the following example: `REDOCLY_CLI_RESPECT_INPUT='{"key":"value","nested":{"nestedKey":"nestedValue"}}' npx @redocly/cli respect test-file.yaml`

---

- --server
- string
- Server overrides for the `sourceDescriptions` object.

>   For example, the following command runs all workflows from the "test-file.yaml" test file and instead of using the server listed in the API description, uses the server at "https://test.com": `npx @redocly/cli respect test-file.yaml --server test=https://test.com`

Reviewer: ```suggestion
  For example, the following command runs all workflows from the "test-file.yaml" Arazzo description and instead of using the server listed in the API description, uses the server at "https://test.com": `npx @redocly/cli respect test-file.yaml --server test=https://test.com`
```

---
slug:
  - /docs/cli/commands/respect
  - /docs/respect/commands/respect
rbac:
  authenticated: read
---

# `respect`

Use this command to execute API tests described in an Arazzo description.

> In addition to the Arazzo specification, `respect` supports specification extensions for API testing: `x-operation` and `x-serverUrl`.

Reviewer 2: ```suggestion
In addition to the Arazzo specification, Respect supports specification extensions for API testing: `x-operation` and `x-serverUrl`.
```

## Usage

```sh
npx @redocly/cli respect <your-test-file | multiple files | files bash query> [-w | --workflow] [-s | --skip] [-v | --verbose] [-i | --input]
```

## Options

{% table %}

- Option {% width="20%" %}
- Type {% width="15%" %}
- Description

---

- -w, --workflow
- [string]
- Workflow names from the test file to run.

>   For example, the following command runs "first-flow" and "second-flow" workflows from the "test-file.yaml" Arazzo description: `npx @redocly/cli respect test-file.yaml --workflow first-flow second-flow`

Reviewer 2: ```suggestion
  For example, the following command runs "first-flow" and "second-flow" workflows from the `test-file.yaml` Arazzo description: `npx @redocly/cli respect test-file.yaml --workflow first-flow second-flow`
```

  {% admonition type="warning" %}
  The `--workflow` option can't be used with `--skip`.
  {% /admonition %}

---

- -s, --skip
- [string]
- Workflow names from the test file to skip.

>   For example, the following command skips the "first-flow" workflow from the "test-file.yaml" Arazzo description: `npx @redocly/cli respect test-file.yaml --skip first-flow`

Reviewer 2: ```suggestion
  For example, the following command skips the "first-flow" workflow from the `test-file.yaml` Arazzo description: `npx @redocly/cli respect test-file.yaml --skip first-flow`
```

Reviewer 2: @JLekawa

  {% admonition type="warning" name="Warning" %}
  The `--skip` option can't be used with `--workflow`.
  {% /admonition %}

---

- -v, --verbose
- boolean
- Runs the command in verbose mode to help with troubleshooting issues.

>   For example, the following command runs all workflows from the "test-file.yaml" Arazzo description in verbose mode: `npx @redocly/cli respect test-file.yaml --verbose`

Reviewer 2: ```suggestion
  For example, the following command runs all workflows from the `test-file.yaml` Arazzo description in verbose mode: `npx @redocly/cli respect test-file.yaml --verbose`
```

Reviewer 2: @JLekawa

---

- --har-output
- string
- Path for the `har` file for saving logs.

>   For example, the following command runs all workflows from the "test-file.yaml" Arazzo description and saves the logs to the "logs.har" file: `npx @redocly/cli respect test-file.yaml --har-output='logs.har'`

Reviewer 2: ```suggestion
  For example, the following command runs all workflows from the `test-file.yaml` Arazzo description and saves the logs to the "logs.har" file: `npx @redocly/cli respect test-file.yaml --har-output='logs.har'`
```

Reviewer 2: @JLekawa

---

- --json-output
- string

> - Path for the JSON file for saving logs. For example, the following command runs all workflows from the "test-file.yaml" Arazzo description and saves the logs to the "logs.json" file:`npx @redocly/cli respect test-file.yaml --json-output='logs.json'`

Reviewer 2: ```suggestion
- Path for the JSON file for saving logs. For example, the following command runs all workflows from the `test-file.yaml` Arazzo description and saves the logs to the "logs.json" file:`npx @redocly/cli respect test-file.yaml --json-output='logs.json'`
```

Reviewer 2: @JLekawa

---

- --input
- string
- Input parameters with values that are mapped to the workflow inputs description.

>   For example, the following command maps the "userEmail" and "userPassword" inputs and values to all workflows in the "test.yaml" Arazzo description: `npx @redocly/cli respect test.yaml --input userEmail=name@redocly.com --input userPassword=12345`.

Reviewer 2: ```suggestion
  For example, the following command maps the "userEmail" and "userPassword" inputs and values to all workflows in the `test.yaml` Arazzo description: `npx @redocly/cli respect test.yaml --input userEmail=name@redocly.com --input userPassword=12345`.
```

Reviewer 2: @JLekawa

  You can even include nested values, as in the following example command that maps the "nestedKey" input and value to all workflows in the "test-file.yaml" Arazzo description: `npx @redocly/cli respect test-file.yaml --input '{"key": "value", "nested": {"nestedKey": "nestedValue"}}'`.
  You can also use an environment variable to set the input, as in the following example: `REDOCLY_CLI_RESPECT_INPUT='{"key":"value","nested":{"nestedKey":"nestedValue"}}' npx @redocly/cli respect test-file.yaml`

---

- --server
- string
- Server overrides for the `sourceDescriptions` object.
  For example, the following command runs all workflows from the "test-file.yaml" Arazzo description and instead of using the server listed in the API description, uses the server at "https://test.com": `npx @redocly/cli respect test-file.yaml --server test=https://test.com`

  You can also pass the server overrides as an environment variable, as in the following example:
  `REDOCLY_CLI_RESPECT_SERVER="test=https://test.com"`

---

- --residency
- string

> - Residency location of Reunite application to use if `login` command was not run before.

Reviewer 2: ```suggestion
- If `login` command was not run before, use residency location of Reunite application.
```

Reviewer 2: According to our styleguide we are supposed to start conditional sentences from the if-clause: https://intranet.redocly.com/content/documentation-style-guide#sentences  however, let me know if my suggestion changes the intended meaning of your original sentence.
